### PR TITLE
Ietf119 hackathon

### PR DESCRIPTION
**Pull request by fl1ger**
Moin!

Here are the changes Dave and myself did during the Hackathon on structures and wording. Would like us authors to confirm them before I merge it.

So long
-Ralf

### CODE REVIEW

File: draft-dnsop-deleg.md [hunk 1]
```diff
@@ -115,9 +115,7 @@ An NS record contains the hostname of the nameserver for the delegated namespace
...
-
-This limitation is a barrier for efficient introduction of new DNS technology. New features come with additional overhead as they are constrained by the intersection of resolver and nameserver functionality. New functionality could be discovered insecurely by trial and error, or negotiated after first connection, which is costly and unsafe.
+In the Domain Name System {{!STD13}}, subdomains within the domain name hierarchy are indicated by delegations to servers which are authoritative for their portion of the namespace.  The DNS records that do this, called NS records, contain hostnames of nameservers, which resolve to addresses.  No other information is available to the resolver. It is limited to connect to the authoritative servers over UDP and TCP port 53. nThis limitation is a barrier for efficient introduction of new DNS technology.
```
Comment (chantra) [inline draft-dnsop-deleg.md]: nit: rogue `n` in 'TCP port 53. nThis limitation'

File: draft-dnsop-deleg.md [hunk 2]
```diff
@@ -149,60 +147,23 @@ NS based delegation supports DNS over UDP and TCP, but does not have the ability
...
 * Able to coexist in the ecosystem: DELEG is defined as being a record in the parent, signifying a zone cut. As a result of that design decision, additional effort and time will be required to deploy DELEG to all levels of the DNS hierarchy, especially when considering the Root zone and TLDs. To support the deployment, DELEG must be able to coexist within the ecosystem with the existing NS based methods of resolution. Testing has been done to show that many deployed resolvers can handle DELEG and NS records side-by-side to enable a rollout. 
+* While DELEG can be used in an unsigned zone, it is recommended to use DNSSEC. The DELEG record must be signed or denied in the parent zone when it is signed. Certain security propperties might not be available and hence certain features only will work when the DELEG record is signed
```
Comment (chantra) [inline draft-dnsop-deleg.md]: nit: end the sentence with a period.
s/propperties/properties/

I find the switch from "The DELEG record must be signed or denied in the parent zone when it is signed." to the next sentence makes it hard to follow. I would preface the last sentence with: "Without DNSSEC, certain security properties might...." just to switch back to the context.

File: draft-dnsop-deleg.md [hunk 3]
```diff
@@ -149,60 +147,23 @@ NS based delegation supports DNS over UDP and TCP, but does not have the ability
...
-
-## DNSSEC is RECOMMENDED
-
-While DNSSEC is RECOMMENDED, unsigned DELEG records may be retrieved in a secure way from trusted, Privacy-enabling DNS servers using encrypted transports.
-
-FOR DISCUSSION: This will lead to cyclical dependencies. A DELEG record can introduce a secure way to communicate with trusted, Privacy-enabling DNS servers. For that, it needs to be DNSSEC signed. 
+# DELEG Record Type
 
-### Preventing downgrade attacks
+The DELEG record wire format is the same as the SVCB record defined in {{?RFC9460}}, but with a different value for the resource record type of TBD. The IANA registry for for further Service Parameter Keys that might be needed also is the one used for SVCB.
```
Comment (chantra) [inline draft-dnsop-deleg.md]: nit: 'IANA registry **for for** further Service'
The sentence does not parse. I feel like something has gone missing.

File: draft-dnsop-deleg.md [hunk 4]
```diff
@@ -149,60 +147,23 @@ NS based delegation supports DNS over UDP and TCP, but does not have the ability
...
 
-A flag in the DNSKEY record is used as a backwards compatible, secure signal to indicate to a resolver that DELEG records are present or that there is an authenticated denial of a DELEG record. Legacy resolvers will ignore this flag and use the DNSKEY as is.
+The SVCB record allows for two types of records, the AliasMode and the ServiceMode. The DELEG record uses both. The Target Name either points to a set of name servers answering for the delegated domain if used in ServiceMode or to an SVCB record in AliasMode that then has to be interpreted further to get to the actual name server. The AliasMode DELEG record might point to another AliasMode SVCB record or a CNAME. In order to not allow unbound indirection of DELEG records the maximum number of indirections, CNAME or AliasMode SVCB is 4. The SvcPriority field either can be 0 for AliasMode or 1 for ServiceMode. Different priorities are not used for DELEG delegations.
```
Comment (chantra) [inline draft-dnsop-deleg.md:156]: Should 4 be recommended? It feels very arbitrary.

Comment (chantra) [reply draft-dnsop-deleg.md:156]: I realize this was already present. I think the question still stand, but we should probably address it separately.

File: draft-dnsop-deleg.md [hunk 5]
```diff
@@ -263,53 +232,67 @@ Some zone owners may wish to use multiple providers to serve their zone, in whic
...
-Construction of a DELEG RR requires knowledge which implies communication between the
-operators of the child and parent zones. This communication is an operational matter not covered by this document.
+Both the AliasMode and ServiceMode records can be returned for the DELEG record from the parent. This is different from the SCVB  {{?RFC9460}} specification and only applies for the DELEG RRSet in the parent.
 
-### Signing DELEG RRs
 
-A DELEG RRset MUST be DNSSEC signed if the zone is signed.
+### Rollout
+
+When introduced, the DELEG and SVCB records might not initially be supported by the DNS root or TLD operators. Zone owners may place these records into their zones before the zones above them have done so. However, doing so is only useful for further delegations down the tree as an SVCB record at the zone apex alone does not indicate a new delegation type. The only way to discover new delegations is with the DELEG record at the parent.
```
Comment (chantra) [inline draft-dnsop-deleg.md]: "Zone owners may place these records into their zones before the zones above them have done so."

Zone owners should always place the records in the child zone before the parent or should would end up having a lame DELEG record at the parent.
A side-note is that it will not be effective until the parent supports DELEG and implements the record.

**Review by chantra — COMMENTED** [review]
There is quite a lot of text moved around, which make it hard to really check the diff in its entirety without losing too much context, but overall it seems good to me. Some nits here and there though.

File: draft-dnsop-deleg.md [hunk 6]
```diff
@@ -147,11 +147,11 @@ NS based delegation supports DNS over UDP and TCP, but does not have the ability
...
-* While DELEG can be used in an unsigned zone, it is recommended to use DNSSEC. The DELEG record must be signed or denied in the parent zone when it is signed. Certain security propperties might not be available and hence certain features only will work when the DELEG record is signed
+* While DELEG can be used in an unsigned zone, it is recommended to use DNSSEC. The DELEG record must be signed or denied in the parent zone when it is signed. Without DNSSEC, certain security properties might not be available and hence certain features only will work when the DELEG record is signed.
 
 # DELEG Record Type
 
-The DELEG record wire format is the same as the SVCB record defined in {{?RFC9460}}, but with a different value for the resource record type of TBD. The IANA registry for for further Service Parameter Keys that might be needed also is the one used for SVCB.
+The DELEG record wire format is the same as the SVCB record defined in {{?RFC9460}}, but with a different value for the resource record type of TBD. For extensions SVCB and DELEG use Service Parameter Keys (SvcParamKeys) and new SvcParamKeys that might be needed also also will use the existing IANA Registry.
```
Comment (chantra) [inline draft-dnsop-deleg.md]: other nit: `also also`

**Review by chantra — COMMENTED** [review]
Thanks. LGTM.

File: draft-dnsop-deleg.md [hunk 7]
```diff
@@ -263,53 +232,67 @@ Some zone owners may wish to use multiple providers to serve their zone, in whic
...
+                    ipv4hint=192.0.2.1 ipv6hint=2001:DB8::1 )
+    example.com.  86400  IN NS     ns1.example.com.
+    ns1.example.com.    86400   IN  A  192.0.2.1
+    ns1.example.com     86400   IN  AAAA    2001:DB8::1
 
-### Rollout
+In this example, the authoritative nameserver is delegating using the same parameters as regular DNS, but the delegation as well as the glue can be signed.
 
-When introduced, the DELEG and SVCB records might not initially be supported by the DNS root or TLD operators. Zone owners may place these records into their zones before the zones above them have done so. However, doing so is only useful for further delegations down the tree as an SVCB record at the zone apex alone does not indicate a new delegation type. The only way to discover new delegations is with the DELEG record at the parent.
+Like in SVCB, DELEG also offer the ability to use the Alias form of delegation. The example below shows an example where example.com is being delegated with a DELEG AliasMode record which can then be further resolved using standard SVCB to locate the actual parameters.
```
Comment (moonshiner) [inline draft-dnsop-deleg.md]: Perhaps "Like with SVCB, DELEG offers the ability to use the Alias form of delegation."

Comment (fl1ger) [reply draft-dnsop-deleg.md]: Thanks Tim. Did changed the text.

File: draft-dnsop-deleg.md (hunk 1 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md]: Thanks. Resolved this.

File: draft-dnsop-deleg.md (hunk 2 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md]: Addressed this. Thanks a lot.

File: draft-dnsop-deleg.md (hunk 4 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md:156]: Keeping it 4 for now.

File: draft-dnsop-deleg.md (hunk 5 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md]: Rewrote that section a bit to address it.

File: draft-dnsop-deleg.md (hunk 6 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md]: Fixed. Thanks a lot.

File: draft-dnsop-deleg.md (hunk 3 above)
Comment (fl1ger) [reply draft-dnsop-deleg.md]: That section has been re written and should be clearer now.
